### PR TITLE
[3.9] bpo-44461: Check early that a pdb target is valid for execution. (GH-27227)

### PR DESCRIPTION
**Pull request by jaraco**
https://bugs.python.org/issue44461

Automerge-Triggered-By: GH:jaraco

### CODE REVIEW

File: Lib/test/test_pdb.py [hunk 1]
```diff
@@ -1632,6 +1632,20 @@ def test_module_without_a_main(self):
         self.assertIn("ImportError: No module named t_main.__main__",
                       stdout.splitlines())
 
+    def test_package_without_a_main(self):
+        pkg_name = 't_pkg'
+        module_name = 't_main'
+        os_helper.rmtree(pkg_name)
```
Comment (ambv) [inline Lib/test/test_pdb.py]: Looks like you're missing an `from test.support import os_helper` import for this line to work.

Comment (jaraco) [reply Lib/test/test_pdb.py]: Looking at the header, `support` was already imported, so I re-used that import.

Comment (jaraco) [reply Lib/test/test_pdb.py]: Ugh. Bad idea. Okay, going with your suggestion ;)

Comment (jaraco) [reply Lib/test/test_pdb.py]: Suggestion doesn't work either. I remember now, between 3.9 and 3.10, test.support was refactored.

**Pull request merged**